algorithms/tree/bst/kth_smallest.py

```diff
@@ -0,0 +1,54 @@
+class Node:
+
+    def __init__(self, val, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+def kth_smallest(root, k):
+    stack = []
+    while root or stack:
+        while root:
+            stack.append(root)
+            root = root.left
+        root = stack.pop()
+        k -= 1
+        if k == 0:
+            break
+        root = root.right
+    return root.val
+
+
+class Solution(object):
```

Comment:
"""
Finds the kth smallest element in a BST.

Parameters:
root (TreeNode): Root node of the BST.
k (int): The kth smallest element to find.

Returns:
int: The kth smallest element in the BST.

Note:
- The BST property ensures that the elements are sorted in ascending order.
"""

Reply:
An example of how the docstring  could be added in the code